kvmagent/kvmagent/test/localstorage_testsuite/test_data_volume_1.py

```
@@ -0,0 +1,70 @@
+        snapshot_path = os.path.join(env.SNAPSHOT_DIR, '%s.qcow2' % uuidhelper.uuid())
+
+        vm_utils.take_snapshot(vm_uuid, vol_uuid, vol_path, snapshot_path)
+        vm_utils.take_volumes_snapshots()
+        self.assertTrue(os.path.isfile(snapshot_path))
+
+
+        vm_utils.stop_vm(vm_uuid)
+        pid = linux.find_vm_pid_by_uuid(vm_uuid)
+        self.assertTrue(not pid, 'vm[%s] vm still running' % vm_uuid)
+
+        virtio_vol_body, virtio_pin = vm_utils.build_virtio_vol_with_iothreadpin(virtio_vol_uuid, virtio_vol_path, virtio_vol_iothread_id, virtio_vol_iothread_pin)
```

Ruff (0.12.2)

`virtio_vol_uuid` may be undefined, or defined from star imports

(F405)